### PR TITLE
Cleanup scores, search results and DocID

### PR DESCRIPTION
**Pull request by donomii**
### What's being changed:

* Don't assign to additionalProperties["score"] inside bm25f, wait for explorer to do it
* Assign correctly to SecondarySortValue
* Do not assign keyword scores to .Dist field
* Remove .Score() from StorObj
* Renamed .docID -> DocID to support testing from outside module, and for consistency with Weaviate's code guidelines
* Merges hybrid results and search results structures
* Adds more tests for hybrid, including specific tests for relativeScoreFusion

This PR is motived by reports from users about issues including scores not appearing, or scores being incorrect.  I was unable to reproduce the exact issues reported, but a look through the code revealed the opportunity for some improvements.

There were some code paths where it was possible for scores to become out of sync or misreported, so this PR reworks our internal handling of scores.  For instance, DocID was not being kept in the same structure as the UUID and Score and other object attributes.  This led to situations where code would try to rebuild the DocID from the UUID, or by attaching an auxillary structure to carry that information.  DocID is now kept in the same structure as the rest of the object attributes, from the moment the structure is created.  DocID is a pointer value because 0x00000 is a valid document id, so there needs to be a way to tell the difference between document #0 and an uninitialised value.

Additionally, sometimes the score was being stored in the .Additional[] property structure, sometimes it was kept in the .Score field.  This patch removes all the places where score was directly assigned to .Additional, and replaces them with .Score.  .Additional is(and was) populated at the end of the request, using values from the result structure.

There were also some places where .SecondarySortValue appeared to be set to the wrong value, or not set at all.  Correcting these did not trigger any tests, so they were probably unused, which is its own problem.

The remainder are some smaller cleanups to match the Weaviate coding style, adding more tests.

Remaining TODO:  we do some extra copying on the search results that we don't need to, so we can save a little bit of memory and time by moving all functions to accept []*search.Result

### Review checklist

- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [X] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: test/acceptance_with_go_client/search_test.go [hunk 1]
```diff
@@ -305,6 +305,65 @@ func TestNearVectorAndObjectAutocut(t *testing.T) {
...
+	creator.WithClassName(className).WithProperties(
+		map[string]interface{}{"contents": []string{
+			"specific",
+			"hybrid",
+			"search",
+			"object",
+		}, "num": 4}).Do(ctx)
+	defer c.Schema().ClassDeleter().WithClassName(className).Do(ctx)
+
+	t.Run("hybrid explainscore 1", func(t *testing.T) {
```
Comment (dirkkul) [inline test/acceptance_with_go_client/search_test.go:325]: maybe have this as an table-driven test?

File: test/acceptance_with_go_client/search_test.go [hunk 2]
```diff
@@ -305,6 +305,65 @@ func TestNearVectorAndObjectAutocut(t *testing.T) {
...
+		results, err := c.GraphQL().Raw().WithQuery(fmt.Sprintf("{Get{%s(hybrid:{query:\"rain nice\", fusionType: rankedFusion, alpha: 0.5, properties: [\"contents\"]}){num _additional { score explainScore id }}}}", className)).Do(ctx)
+
```
Comment (dirkkul) [inline test/acceptance_with_go_client/search_test.go:327]: AFAIK the newest go client can do this with the builder pattern - I assume you copied this from another test that was written when the go client did not support this yet

**Review by dirkkul — APPROVED** [review]
Did not look at every change in detail, but this contains nothing scary :)

File: test/acceptance_with_go_client/search_test.go (hunk 2 above)
Comment (donomii) [reply test/acceptance_with_go_client/search_test.go:327]: Will fix in another patch

File: test/acceptance_with_go_client/search_test.go (hunk 1 above)
Comment (donomii) [reply test/acceptance_with_go_client/search_test.go:325]: I'll try that when I install goland :)

**Review by dirkkul — APPROVED** [review]

**Pull request merged**